### PR TITLE
Show available languages on Course Overview page

### PR DESCRIPTION
**Pull request by dmcavoy**
We do not currently show the list of languages a UnitGroup is available in on the Course Overview page. We created a way to determine if a UnitGroup is translated into a language for the Curriculum Catalog (if all the units in the UnitGroup are translated). This PR leverages that to show the available languages on the Course Overview page.

Screenshot 2023-09-06 at 9 28 22 PM

Note: In looking into this I realized that in other languages we are not marking a Course Version as recommended currently and I have not been able to hunt down the issue.  Teacher Tools can you track this?

Screenshot 2023-09-06 at 9 30 34 PM

## Links

Slack Conversation: https://codedotorg.slack.com/archives/CFTFD6BPV/p1692043933731759
Jira: https://codedotorg.atlassian.net/browse/TEACH-663

## Testing story

I tested it locally.

### CODE REVIEW

File: dashboard/app/models/unit_group.rb [hunk 1]
```diff
@@ -500,6 +500,22 @@ def self.latest_stable_version(family_name, locale: 'en-us')
     end
   end
 
+  def supported_locale_codes
+    locales = default_unit_group_units.first&.script&.supported_locales || []
```
Comment (hannahbergam) [inline dashboard/app/models/unit_group.rb:504]: To make sure I'm following- does the supported_locales list in the first unit represent the translations of the rest of its units here?

**Review by hannahbergam — COMMENTED** [review]
LGTM! One small question

File: dashboard/app/models/unit_group.rb (hunk 1 above)
Comment (etaderhold) [reply dashboard/app/models/unit_group.rb:504]: This line just gets the locales for the first unit. The next line checks that those locales are also present in the rest of the units...but I notice that the result of that `filter` call isn't assigned back to the `locales` array. Seems like a bug. I should write a test to make sure.

Comment (hannahbergam) [reply dashboard/app/models/unit_group.rb:504]: Ahh thank you for that. I would need to know more about ruby to understand if this works in-place.

**Review by hannahbergam — APPROVED** [review]
LGTM!

**Pull request merged**